src/aiida/orm/nodes/data/code/legacy.py

```diff
@@ -75,13 +75,15 @@ class Code(AbstractCode):
     def __init__(self, remote_computer_exec=None, local_executable=None, input_plugin_name=None, files=None, **kwargs):
         super().__init__(**kwargs)
 
-        warn_deprecation(
-            'The `Code` class is deprecated. To create an instance, use the '
-            '`aiida.orm.nodes.data.code.installed.InstalledCode` or `aiida.orm.nodes.data.code.portable.PortableCode` '
-            'for a "remote" or "local" code, respectively. If you are using this class to compare type, e.g. in '
-            '`isinstance`, use `aiida.orm.nodes.data.code.abstract.AbstractCode`.',
-            version=3,
-        )
+        if getattr(self, '_EMIT_CODE_DEPRECATION_WARNING', True):
+            warn_deprecation(
+                'The `Code` class is deprecated. To create an instance, use the '
```

Comment:
This warning might be a bit confusing if it's printed when user tries to create Code via CLI. At the same time the text is already long. :shrug: